closed/src/java.base/zos/native/libsyslookup/syslookup.c

```diff
@@ -0,0 +1,325 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2024, 2024 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+/* We need to skip IBM_ATOE defined in src/java.base/zos/native/libwrappers/include
+ * to generate the dynamic library due to the conflicts with /usr/inlcude.
+ */
+#ifdef IBM_ATOE
+#undef IBM_ATOE
+#endif /* IBM_ATOE */
+
+#ifndef _AIX_COMPATIBILITY
+#define _AIX_COMPATIBILITY
+#endif /* _AIX_COMPATIBILITY */
```

Comment:
Why do we want `_AIX_COMPATIBILITY` on z/OS?

Reply:
This is simply a symbol defined as part of dependencies required in <math.h>
(See https://www.ibm.com/docs/sk/zos/2.4.0?topic=functions-finite-determine-infinity-classification-floating-point-number).
Removing it will end up with a bunch of errors with undeclared identifiers detected by the compiler on z/OS.
```
ERROR CCN3045 ./syslookup.c:218   Undeclared identifier finite.
```

Reply:
A comment is added as suggested.